sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/checks/IndentationCheck.java

```diff
@@ -154,18 +158,64 @@ private int startIndent(Node node) {
     return indent;
   }
 
-  private void checkClosingTag(Node node) {
-    XmlTextRange startLocation = XmlFile.startLocation((Element) node);
-    XmlTextRange endLocation = XmlFile.endLocation((Element) node);
-    boolean isSelfClosing = startLocation.getEndLine() == endLocation.getEndLine()
-      && startLocation.getEndColumn() == endLocation.getEndColumn();
-    if (!isSelfClosing && startLocation.getEndLine() != endLocation.getStartLine()) {
-      int startIndent = startIndent(node.getPreviousSibling());
-      int endIndent = startIndent(node.getLastChild());
+  private void checkClosingTag(Element element) {
+    XmlTextRange startLocation = XmlFile.startLocation(element);
+    XmlTextRange endLocation = XmlFile.endLocation(element);
+    if (!isSelfClosing(element) && startLocation.getEndLine() != endLocation.getStartLine()) {
+      if (!needToCheckIndentation(element)) {
+        return;
+      }
+      int startIndent = startIndent(element.getPreviousSibling());
+      int endIndent = startIndent(element.getLastChild());
       if (startIndent != endIndent) {
         reportIssue(endLocation, startIndent);
       }
     }
   }
+
+  private boolean needToCheckIndentation(Element element) {
+    // When tag is inside lines of text we do not check indentation
+
+    if (element.getChildNodes().getLength() > 1) {
+      return true;
+    }
+
+    Node previous = element.getPreviousSibling();
+    if (isNonEmptyTextNode(previous)) {
+      return false;
+    }
+
+    Node next = element.getNextSibling();
+    if (isNonEmptyTextNode(next)) {
+      return false;
+    }
+
+    // Current tag can be encapsulated in another tag which is inside lines of text
+    for (Node parent = element.getParentNode(); parent != null && parent.getChildNodes().getLength() == 1; parent = parent.getParentNode()) {
+      short parentType = parent.getNodeType();
+      if (parentType == Node.ELEMENT_NODE) {
+        Node parentPrev = parent.getPreviousSibling();
+        if (null != parentPrev && parentPrev.getNodeType() == Node.TEXT_NODE) {
```

Comment:
why not reusing the `isNonEmptyTextNode()` method?
It seems to me that you can simplify this part quite a lot:
```java
if (isNonEmptyTextNode(parent.getPreviousSibling())) {
  return false;
}
```